File: JAVA/2024-12-31/120876.java
public class Main {
  public static void main(String[] args) {
    }
}

class solution
{
  public int solution(int[][] lines) {
    {
      HashMap<Integer, Integer> map = new HashMap<>();
      for (int[] line : lines)
      {
        for (int i = line[0]; i < line[1]; i++)
        {
          map.put(i, map.getOrDefault(i, 0) + 1);
        }
      }
      int result = 0;
      for (int n : map.values()) {
        if (n > 1)
        {
          result++;
        }
      }
      return result;
    }
}
// 이 문제는 사실 시작하지 못하고 찾아서 가져온 코드이다.
// 문제는 이해했지만, 겹치는 부분을 어떻게 계산해줘서 해야할지 감이 잡히지 않았기 때문에 다시 풀어봐야할 문제이다.

File: C++/2024-12-31/120876.cpp
#include <string>
#include <vector>
using namespace std;

int solution(vector<vector<int>> lines) 
{
    int answer = 0;
	vector<int> line(201, 0);

	for (int i = 0; i < lines.size(); i++)
	{
		int s = lines[i][0];
		int e = lines[i][1];

		for (int j = s; j < e; j++)
		{
			line[j + 100]++;
		}
	}

	for (int i = 0; i < line.size(); i++)
	{
		if (line[i] > 1)
			answer++;
	}

    return answer;
}
